### ISSUE TITLE
Points layer dimensions are not permuted when converted

### ISSUE THREAD

**Issue by gselzer** [imagej/napari-imagej#302]
I tried writing a SciJava script that would create a mesh from a grayscale image, using the value(s) at point(s) described within a `Points` layer. This script was altered from a similar script @elevans sent me:
```jython
#@ OpService ops
#@ Img img
#@ net.imglib2.roi.geom.real.RealPointCollection isolevels
#@output net.imagej.mesh.Mesh output

from net.imglib2.type.logic import BitType

def apply_isolevel(image, isolevels):
    """Apply the desired isolevel on the input image.

    Apply a desired isolevel (i.e. isosurface) value on the input image,
    returning a BitType image that can be used with the marching cubes Op.

    :param image:

        Input ImgPlus.

    :param isolevel:

        Points corresponding to the locations that should be isolevels

    :return:

        An ImgLib2 Mesh at the specified isolevel.
    """
    ra = img.randomAccess()
    isolevel = image.firstElement().copy()
    for point in isolevels.points():
        for i in range(3):
            ra.setPosition(int(point.getDoublePosition(i)), i)
        isolevel.add(ra.get())

    isolevel.set(isolevel.get() / isolevels.size())
    print(isolevel)
    print(img.dimensionsAsLongArray())

    bin_img = ops.create().img(image, BitType())
    ops.threshold().apply(bin_img, image, isolevel)

    return ops.geom().marchingCubes(bin_img)
```

This script doesn't work as intended, as `pyimagej` is smart enough to permute the dimensions of the image, but `napari-imagej` does not permute the dimensions of the points.

@elevans @ctrueden how do you think we should proceed? If we decide that we want to permute the points in the same way, does pyimagej expose that functionality?

**Comment by ctrueden** [comment]
> If we decide that we want to permute the points in the same way, does pyimagej expose that functionality?

I figure we should be permuting the points, yeah. But I don't understand your question. The `_mesh_to_surface` and `_surface_to_mesh` functions of napari-imagej are where these conversions happen, no? So don't you just need to take care of it there?

**Comment by gselzer** [comment]
My question involves the fact that I do not know the dimensionality of the points beforehand, so there will have to be some case logic to differentiate (2-d points vs. 3-d vs. ...) - is there any exposed utility in PyImageJ to tell me how to permute the dimensions, or do I need to write this case logic explicitly in napari-imagej?

**Comment by ctrueden** [comment]
The points are always 3D, aren't they? Meshes are 2D manifolds embedded in 3D space.

**Comment by gselzer** [comment]
Note for future self: Ensure that the dimensions are swapped for Shapes layers as well. In napari-imagej v0.1.0, I tried the following:
* Opened the napari-imagej and the Fiji UIs simultaneously
* Open an image in the Fiji window, and draw a ROI on it. I drew an ellipse with high eccentricity in the top right corner.
* Transfer the image to napari. The ROI comes with but the dimensions are wrong.
We should make sure that this issue has since been resolved - it may have been earlier with the advanced data transfer buttons.
